modules/json/JsonNatives.cpp

```diff
@@ -0,0 +1,938 @@
+{
+	auto value = params[1];
+	if (!JsonMngr->IsValidHandle(value))
+	{
+		MF_LogError(amx, AMX_ERR_NATIVE, "Invalid JSON value! %d", value);
+		return -1;
+	}
+
+	JS_Handle parent;
+	auto result = JsonMngr->GetValueParent(value, &parent);
+
+	return (result) ? parent : -1;
+}
+
+//native JSONType:json_get_type(const JSON:value);
+static cell AMX_NATIVE_CALL amxx_json_get_type(AMX *amx, cell *params)
+{
+	auto value = params[1];
+	if (!JsonMngr->IsValidHandle(value))
+	{
+		MF_LogError(amx, AMX_ERR_NATIVE, "Invalid JSON value! %d", value);
+		return static_cast<cell>(JSONTypeError);
+	}
+
+	return JsonMngr->GetHandleJSONType(value);
+}
+
+//native JSON:json_init_object();
+static cell AMX_NATIVE_CALL amxx_json_init_object(AMX *amx, cell *params)
+{
+	JS_Handle handle;
+	auto result = JsonMngr->InitObject(&handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_init_array();
+static cell AMX_NATIVE_CALL amxx_json_init_array(AMX *amx, cell *params)
+{
+	JS_Handle handle;
+	auto result = JsonMngr->InitArray(&handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_init_string(const value[]);
+static cell AMX_NATIVE_CALL amxx_json_init_string(AMX *amx, cell *params)
+{
+	int len;
+	JS_Handle handle;
+	auto result = JsonMngr->InitString(MF_GetAmxString(amx, params[1], 0, &len), &handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_init_number(value);
+static cell AMX_NATIVE_CALL amxx_json_init_number(AMX *amx, cell *params)
+{
+	JS_Handle handle;
+	auto result = JsonMngr->InitNum(params[1], &handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_init_real(Float:value);
+static cell AMX_NATIVE_CALL amxx_json_init_real(AMX *amx, cell *params)
+{
+	JS_Handle handle;
+	auto result = JsonMngr->InitNum(amx_ctof(params[1]), &handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_init_bool(bool:value);
+static cell AMX_NATIVE_CALL amxx_json_init_bool(AMX *amx, cell *params)
+{
+	JS_Handle handle;
+	auto result = JsonMngr->InitBool(params[1] != 0, &handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_init_null();
+static cell AMX_NATIVE_CALL amxx_json_init_null(AMX *amx, cell *params)
+{
+	JS_Handle handle;
+	auto result = JsonMngr->InitNull(&handle);
+
+	return (result) ? handle : -1;
+}
+
+//native JSON:json_deep_copy(const JSON:value);
+static cell AMX_NATIVE_CALL amxx_json_deep_copy(AMX *amx, cell *params)
+{
+	auto value = params[1];
+	if (!JsonMngr->IsValidHandle(value))
+	{
+		MF_LogError(amx, AMX_ERR_NATIVE, "Invalid JSON value! %d", value);
+		return -1;
+	}
+
+	JS_Handle clonedHandle;
+	auto result = JsonMngr->DeepCopyValue(value, &clonedHandle);
+
+	return (result) ? clonedHandle : -1;
+}
+
+//native bool:json_free(&JSON:value);
+static cell AMX_NATIVE_CALL amxx_json_free(AMX *amx, cell *params)
+{
+	auto value = MF_GetAmxAddr(amx, params[1]);
+	if (!JsonMngr->IsValidHandle(*value))
+	{
+		MF_LogError(amx, AMX_ERR_NATIVE, "Invalid JSON value! %d", *value);
```

Comment:
I'm not sure erroring on `json_free` is useful. I believe others APIs don't do that as well.

Reply:
Gonna be removed.

Reply:
Done along with comments.